### PR TITLE
Fetch config from the environment.

### PR DESCRIPTION
**Pull request by zenhack**
piggy-backed on top of #36.

From here I'll factor out the http/tls stuff into a package that we can pull in from hil-vpnd.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,24 +11,24 @@ change the explanations to avoid this prerequisite in the future.
...
-
-	"host=localhost port=5432 user=username password=pass dbname=obmd"
+OBMd pulls configuration information from environment variables; The
+following variables are recognized:
+
+* `DB_TYPE` -- the DBMS to use for storage. This can be either `sqlite3`
+  or `postgres`.
+* `DB_PATH` -- how to connect to the database.
+  * For `sqlite3` this can be either a file path or the special string
+    `:memory:` for an in-memory database. For
```
Comment (naved001) [inline README.md]: extra `For` at the end.

**Comment by naved001** [comment]
Does approving this means I approved all the other commits that were a part of this? :open_mouth:

**Comment by zenhack** [comment]
@naved001, I'm going to assume we treat those as part of the earlier PRs, and don't merge this until those land.

File: README.md (hunk 1 above)
Comment (zenhack) [reply README.md]: Fixed.

**Comment by zenhack** [comment]
Rebased; assuming travis still passes I'll hit merge.

**Comment by zenhack** [comment]
CI passed, merging.

**Pull request merged**